apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.usecase.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { EnvironmentRepository, NotificationTemplateRepository } from '@novu/dal';
+import { DeleteWorkflowCommand } from './delete-workflow.command';
+import { ControlValuesRepository, MessageTemplateRepository } from '@novu/dal/src';
+import { WorkflowNotFoundException } from '../../exceptions/workflow-not-found';
+
+@Injectable()
+export class DeleteWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private msgTemplateRepo: MessageTemplateRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+
+  async execute(command: DeleteWorkflowCommand): Promise<void> {
+    await this.validateEnvironment(command);
+    const workflow = await this.findById(command);
+    if (!workflow) {
+      throw new WorkflowNotFoundException(command.workflowId);
+    }
+    await this.deleteComponents(command, workflow);
+  }
+
+  private async deleteComponents(command: DeleteWorkflowCommand, workflow) {
+    await this.notificationTemplateRepository.delete(buildDeleteQuery(command));
+    await this.deleteMsgTemplatesIfRequired(workflow, command);
+    this.controlValuesRepository.de({});
+  }
+
+  private async findById(command: DeleteWorkflowCommand) {
+    return await this.notificationTemplateRepository.findByIdQuery({
+      id: command.workflowId,
+      environmentId: command.user.environmentId,
+    });
+  }
+
+  private async deleteMsgTemplatesIfRequired(workflow, command: DeleteWorkflowCommand) {
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** As above, please use the full name rather than abbreviations.‏